sematic/config/server_settings.py

```diff
@@ -42,144 +39,72 @@ class ServerSettingsVar(enum.Enum):
     GRAFANA_PANEL_URL = "GRAFANA_PANEL_URL"
 
 
-class MissingServerSettingsError(Exception):
-    def __init__(self, missing_settings: ServerSettingsVar):
-        # TODO #264: this bleeds cli implementations details
-        # this message should be set when triaging all exceptions before being surfaced
-        # to the user
-        message = """
-Missing settings: {}
-
-Set it with:
-
-    $ sematic server-settings set {} VALUE
-""".format(
-            missing_settings.value, missing_settings.value
-        )
-        super().__init__(message)
-
-
-ServerSettings = Dict[ServerSettingsVar, str]
-# global settings cache
-_settings: Optional[ServerSettings] = None
-
-
-def _get_default_server_settings() -> ServerSettings:
-    """
-    Returns modifiable default settings.
-    """
-    return {}
-
-
-def _get_server_settings_file() -> str:
-    """
-    Returns the path to the server settings file according to the configuration.
-    """
-    return os.path.join(get_config_dir(), SERVER_SETTINGS_FILE)
-
-
-def _load_server_settings() -> ServerSettings:
-    """
-    Loads the settings from the configured settings file.
-    """
-    raw_settings = _load_settings(_get_server_settings_file())
-
-    if raw_settings is None:
-        return _get_default_server_settings()
-
-    server_settings = {}
-
-    for var, value in raw_settings.items():
-        normalized_var = _normalize_enum(ServerSettingsVar, var)
+class ServerSettingsScope(SettingsScope):
+    @property
+    def settings_file_path(self) -> str:
+        """
+        Temporary override to enable smooth transition from one file
+        to one file per scope.
 
-        if normalized_var is None:
-            raise ValueError(f"Unknown server setting {var}!")
+        TODO: Remove this code.
+        """
+        expected_path = super().settings_file_path
 
-        server_settings[normalized_var] = str(value)
+        user_settings_file_path = get_user_settings_file_path()
+        if not os.path.isfile(expected_path) and os.path.isfile(
+            user_settings_file_path
+        ):
```

Comment:
This avoids an extra I/O operation on the happy path.
```suggestion
        if not os.path.isfile(expected_path):
          user_settings_file_path = get_user_settings_file_path()
          if os.path.isfile(user_settings_file_path):
```